### ISSUE TITLE
Requested outputs

### ISSUE THREAD

**Issue by matlipson** [21centuryweather/RNS_Sydney_1km#5]
From @andrewbrown31  email (Date: Thursday, 12 December 2024 at 2:51 PM)

### What outputs do you need?
If possible, we would like to have the following single-level variables at hourly instantaneous (unless otherwise noted) output (or even 10-minute if possible):
[10 m U wind on B grid](http://reference.metoffice.gov.uk/um/stash/m01s03i225)
[10 m V wind on B grid](http://reference.metoffice.gov.uk/um/stash/m01s03i226)
[Near surface temperature](http://reference.metoffice.gov.uk/um/stash/m01s03i236)
[Near surface specific humidity](http://reference.metoffice.gov.uk/um/stash/m01s03i237)
[Precipitation](http://reference.metoffice.gov.uk/um/stash/m01s05i216) (mean rate)
Surface radiation
[Latent heat flux](http://reference.metoffice.gov.uk/um/stash/m01s03i234),
S[ensible heat flux](http://reference.metoffice.gov.uk/um/stash/m01s03i217)
[Net down surface SW](http://reference.metoffice.gov.uk/um/stash/m01s01i202)
[Net down surface LW](http://reference.metoffice.gov.uk/um/stash/m01s02i201)
[Boundary layer height](http://reference.metoffice.gov.uk/um/stash/m01s00i025)
 
We’d also like 3D U and V wind (3 hourly or even hourly if possible), but we would prefer having model level (rho grid) output up to a height of 6 km, rather than standard pressure level output. What are your thoughts? We would need the corresponding heights, so potentially geopotential height would also need to be saved, unless the rho grid can be processed onto a regular height grid in the stash output.
 
In terms of hub heights, usually 100 m or 200 m are used, but future heights could be up to 300 m. So, one option could be to save hourly/10-minute U and V winds at those levels if possible (or at the closest model levels to those heights). Another option could be that we interpolate to those heights as part of our own post-processing/analysis, which would be just as accurate if we can save the model level data (and still fairly accurate with standard pressure levels, although in that case geopotential height would also need to be saved on pressure levels).
 
Happy to discuss further by email or meeting. @Claire Vincent please let me know if I missed anything out.
 
Cheers, Andrew

**Comment by matlipson** [comment]
Does anyone know if NRI/ others have a good standard STASH pack (maybe for AUS2200)? I usually use stashpack1, then some additional outputs in a custom stashpack6, which includes most of your requested variables. The only thing I'd change from the standard stashpack1 is to change latent and sensible heat fluxes from instantaneous to hourly average (as that's the typical observation period, instantaneous is far too noisy for obs).

**Comment by matlipson** [comment]
Two updates: 

I have completed four weeks of simulations across 6 experiments, each with 4 domains (all include the "Large" domain). A list of experiments is [here](https://github.com/21centuryweather/RNS_Sydney_1km?tab=readme-ov-file#experiments).

I have also documented the 94 outputs [here](https://github.com/21centuryweather/RNS_Sydney_1km?tab=readme-ov-file#outputs). I am in the process of converting the raw UM outputs to compressed netcdf.  This will take some time as it requires manual definitions. If you want access to these, please let me know your priorities.

**Comment by andrewbrown31** [comment]
Awesome work Mat, the documentation in this repository will be very useful, both for analysing output from these runs as well as helping others run the RNS (including me).

**Comment by matlipson** [comment]
I'm guessing might be some issues that I haven't considered (e.g. pointing to my data storage ce10 that you won't have access to). So perhaps when I'm next in Melbourne (early Feb) we can sit down together and try getting stuff running for your use cases?

**Comment by andrewbrown31** [comment]
That would be good. I was already able to get the u-dl705/u-dg768 (ostia-enabled) workflow going for the [CAPE-k domain](https://github.com/21centuryweather/RNS_CAPE-K_1km), by following your steps (creating an /opt file for the new domain). That was with all the standard options in u-dg768. What might be useful is helping with the stash output, as I haven't found this straightforward to change.

**Comment by andrewbrown31** [comment]
@matlipson is it correct that all the output requested here has been added to stashpack 6, other than the following?

> We’d also like 3D U and V wind (3 hourly or even hourly if possible), but we would prefer having model level (rho grid) output up to a height of 6 km, rather than standard pressure level output

If so I have put a solution to this in #8, so we can potentially close both issues once that gets into your suite

**Comment by matlipson** [comment]
Yep! All of those outputs are included (some are in stashpack1, e.g. precip). 

Great to hear you have a solution for model level output. Would it be preferable to have a height output rather than model/rho? I know that's possible, I would just have to look into it.

**Comment by andrewbrown31** [comment]
Great! 

Probably output on regular height intervals would be preferable, although the rho levels aren't too bad to work with (i.e. they can be easily interpolated in post-processing/analysis scripts). I had a bit of a look into the height output in #8 if that helps.
